src/UI/Components/TreeTable/Helpers/TreeTableHelper.ts

```diff
@@ -44,13 +44,13 @@ export abstract class BaseTreeTableHelper<A extends AttributeTree>
   createRows(
     expansionState: ExpansionState,
     setState: (state: ExpansionState) => void
-  ): TreeRow[] {
+  ): { rows: TreeRow[]; openAll: () => void; closeAll: () => void } {
     const createOnToggle = (key: string) => () =>
```

Comment:
Does this not work?
`const createOnToggle = (key: string) => setState(this.expansionManager.toggle(expansionState, key));`

Reply:
As fas as I understand this piece of code, not really, as it is used in creator when that key is used and returns a function that doesn't require any key/path to toggle visibility, I mean it could work, but then we would just move that anonymous function somewhere else

Reply:
Hmm, it's a pattern they are reusing a lot... returning anonymous functions. Fine, I guess it can stay like this since it would require too much logic or adjustment to avoid having this kind of situations.